api/api_test.go

```diff
@@ -0,0 +1,103 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/conduit/conduit/pipeline"
+)
+
+type mockStatusProvider struct {
+	s   pipeline.Status
+	err error
+}
+
+func (m mockStatusProvider) Status() (pipeline.Status, error) {
+	if m.err != nil {
+		return pipeline.Status{}, m.err
+	}
+	return m.s, nil
+}
+
+func TestStartServer_BadAddress(t *testing.T) {
+	l, h := test.NewNullLogger()
+	// capture the fatal log if any.
+	l.ExitFunc = func(int) {}
+	sp := mockStatusProvider{}
+
+	shutdown, err := StartServer(l, sp, "bad address")
+	defer shutdown(context.Background())
+	require.NoError(t, err)
+	time.Sleep(1 * time.Millisecond)
+
+	require.Len(t, h.Entries, 1)
+	require.Equal(t, h.LastEntry().Level, logrus.FatalLevel)
+}
+
+func TestStartServer_GracefulShutdown(t *testing.T) {
+	l, h := test.NewNullLogger()
+	// capture the fatal log if any.
+	l.ExitFunc = func(int) {}
+	sp := mockStatusProvider{}
+	shutdown, err := StartServer(l, sp, "bad address")
```

Comment:
Per other discussion, add a 2nd shutdown attempt to verify no panic.

Reply:
It was definitely exiting / calling the `Fatalf` log. Thats why `ExitFunc` was needed for this test.